code/modules/clothing/masks/gasmask.dm

```diff
@@ -100,13 +100,13 @@
 	actions_types = list(/datum/action/item_action/toggle_mask, /datum/action/item_action/change_appearance_mask, /datum/action/item_action/toggle_voicechanger)
 	species_fit = list(VOX_SHAPED, GREY_SHAPED)
 	var/list/clothing_choices = list()
-
+//someone who isn't me should really refactor this so it isnt instantiating all the masks in the game on startup
 /obj/item/clothing/mask/gas/voice/New()
 	..()
-	for(var/Type in existing_typesof(/obj/item/clothing/mask) - list(/obj/item/clothing/mask, /obj/item/clothing/mask/gas/voice))
-		clothing_choices += new Type
-	return
-	
+	if(type == /obj/item/clothing/mask/gas/voice)
+		for(var/Type in existing_typesof(/obj/item/clothing/mask) - list(/obj/item/clothing/mask, /obj/item/clothing/mask/gas/voice))
```

Comment:
N O
This fixes the runtime/infinite loop that happens when a subtype spawns. Or something similar. You talked with Sood about this 1 hour ago.